leveldown.js

```diff
@@ -83,7 +83,8 @@ SubDown.prototype._open = function (opts, cb) {
     return done()
   }
 
-  this.db.on('open', this.open.bind(this, opts, done))
+  this.db.once('open', this.open.bind(this, opts, done))
+  this.db.open()
 
```

Comment:
This fixes the open, close, open, close test in `abstract-leveldown` (which @finnp commented out in previous PR)

Reply:
Maybe we should tweak it to do `if (this.db.isClosed()) { this.db.open() }`?

Reply:
@ralphtheninja would this work?

```js
SubDown.prototype._open = function (opts, cb) {
  var self = this

  this.db.open(function (err) {
    if (err) return cb(err)

    var subdb = down(self.db, 'subleveldown')

    if (subdb && subdb.prefix) {
      self.prefix = subdb.prefix + self.prefix
      self.leveldown = down(subdb.db)
    } else {
      self.leveldown = down(self.db)
    }

    if (self._beforeOpen) self._beforeOpen(cb)
    else cb()
  })
}
```

Reply:
@ralphtheninja ping in case you missed this because GH hides it

Reply:
Much cleaner! Let me try it out.

Reply:
Updated.